npc.cpp

```diff
@@ -89,23 +89,32 @@ void Game::GrowWMons(int rate)
 								 (ItemDefs[mon].flags & ItemType::DISABLED)))
 								avail = 1;
 
-							if (avail)
+							if (avail) {
 								wanted += TerrainDefs[reg->type].wmonfreq;
+							}
 						}
 					}
 				}
 
 				wanted /= 10;
 				wanted -= mons;
 				wanted = (wanted*rate + getrandom(100))/100;
+
 				if (wanted > 0) {
-					for (int i=0; i< wanted;) {
-						int m=getrandom(8);
-						int n=getrandom(8)*2+m%2;
+					// TODO: instead of loop guard need to check how many available regions
```

Comment:
@sgb To make it non get random, it will require changing also `MakeWMon` function, i will make it but as another PR

Reply:
Why would MakeWMon need to change?